src/main/kotlin/dev/sparky200/skriptnbt/nbt/NbtCompound.kt

```diff
@@ -0,0 +1,95 @@
+package dev.sparky200.skriptnbt.nbt
+
+import org.jetbrains.annotations.ApiStatus.Experimental
+
+@Experimental
+class NbtCompound(val resolver: NbtTagResolver = NbtTagResolver.Default) : NbtTag<Map<String, Any?>> {
+    private val map = mutableMapOf<String, NbtTag<*>>()
+
+    override val value: Map<String, Any?>
+        get() = map.entries.associate { (key, value) -> key to value.value }
+
+    val keys: Collection<String>
+        get() = map.keys
+
+    // TODO: type it!
+    operator fun set(key: String, value: Any?) {
+        map[key] = resolver.resolve(value)
+    }
+
+    fun setTag(key: String, tag: NbtTag<*>) {
+        map[key] = tag
+    }
+
+    operator fun get(key: String): Any? = map[key]?.value
+    fun getAsTag(key: String): NbtTag<*>? = map[key]
+
+    inline fun <reified T> getAs(key: String): T? {
+        val tag = getAsTag(key)
+        if (tag is T) return tag
+        if (tag != null && tag.value is T) return tag.value as T
+        return null
+    }
+
+    fun clear() {
+        map.clear()
+    }
+
+    fun remove(key: String): Any? {
+        return map.remove(key)?.value
+    }
+
+    operator fun contains(key: String): Boolean = key in map
+
+    override fun toNbtString(): String = map.entries.joinToString(separator = ",", prefix = "{", postfix = "}") { (key, value) -> "$key=${value.toNbtString()}" }
+
+    override fun toString(): String {
+        return "NbtCompound{${map.entries.joinToString { "\"${it.key}\"=\"${it.value}\"" }}}"
+    }
+}
+
+fun NbtCompound(resolver: NbtTagResolver, vararg elements: Pair<String, Any?>): NbtCompound {
+    val compound = NbtCompound(resolver)
+    elements.forEach { (key, value) ->
+        compound[key] = value
+    }
+    return compound
+}
+
+fun NbtCompound(vararg elements: Pair<String, Any?>): NbtCompound = NbtCompound(NbtTagResolver.Default, *elements)
+
+private fun parseArrayOrListTag(value: String): NbtTag<*> {
+    TODO()
+}
+
+private fun parseStringTag(value: String): NbtTag<*> {
+    if (value.startsWith("\"")) {
+        if (!value.endsWith("\"")) throw IllegalArgumentException("string literal must end with \"")
+        return NbtStringTag(value.substring(1, value.length - 1))
+    }
```

Comment:
Quotes in stringified NBT can be single `'` or double `"`!
I think both options are used to make escaping easier.
<img width="851" alt="image" src="https://github.com/user-attachments/assets/7dbcca4e-80b2-47fc-b2bf-4838450d05da">

Reply:
Oh I guess you'll need to account for the escaped `"` or `'` depending on which outer bracket was used in the string.